src/com/teamcautionrobotics/robot2018/Lift.java

```diff
@@ -0,0 +1,173 @@
+package com.teamcautionrobotics.robot2018;
+
+import edu.wpi.first.wpilibj.Encoder;
+import edu.wpi.first.wpilibj.VictorSP;
+import edu.wpi.first.wpilibj.PIDController;
+import edu.wpi.first.wpilibj.PIDSource;
+import edu.wpi.first.wpilibj.PIDSourceType;
+
+public class Lift implements PIDSource {
+
+    enum LiftLevel {
+        GROUND(0), SWITCH(19), LOW_SCALE(48), HIGH_SCALE(76);
+
+        private static LiftLevel[] values = values();
+
+        double height;
+
+        private LiftLevel(double height) {
+            this.height = height;
+        }
+
+        public LiftLevel next() {
+            if (this.ordinal() == values.length - 1) {
+                return this;
+            } else {
+                return values[this.ordinal() + 1];
+            }
+        }
+
+        public LiftLevel previous() {
+            if (this.ordinal() == 0) {
+                return this;
+            } else {
+                return values[this.ordinal() - 1];
+            }
+        }
+    }
+
+    private VictorSP liftMotor;
+    private Encoder liftEncoder;
+    private PIDController pidController;
+
+    private double destinationHeight;
+
+    public Lift(int motorPort, int encoderChannelA, int encoderChannelB, double Kp, double Ki,
+            double Kd) {
+        liftMotor = new VictorSP(motorPort);
+        liftEncoder = new Encoder(encoderChannelA, encoderChannelB);
+        liftEncoder.setDistancePerPulse((4 * Math.PI) / 1024);
+        pidController = new PIDController(Kp, Ki, Kd, 0, this, this::move);
+        pidController.setOutputRange(-1, 1);
+        pidController.setAbsoluteTolerance(3);
+    }
+
+    /**
+     * @param power positive is ascending, negative is descending, range of [-1, 1]
+     */
+    public void move(double power) {
+        liftMotor.set(power);
+    }
+
+    public void stop() {
```

Comment:
I think you forgot to remove this. I think I explained my reasoning well, but if not, I will write it upon your request.